Sources/NetworkService/Request.swift

```diff
@@ -0,0 +1,39 @@
+import Foundation
+
+/// Encapsulates an HTTP request with the necessary components.
+public struct Request {
+    public typealias HTTPHeaderKey = String
+    public typealias HTTPHeaderValue = String
+    public typealias HTTPHeaders = [HTTPHeaderKey: HTTPHeaderValue]
+
+    /// Creates a request with the provided parameters.
+    ///
+    /// - Parameters:
+    ///    - URI: The URL of the request.
+    ///    - requestMethod: The HTTP request method.
+    ///    - headers: HTTP header fields for the request.
+    ///    - body: The data sent as the message body of the request.
+    public init(
+        URI: URL,
+        requestMethod: HTTPMethod,
+        headers: Request.HTTPHeaders,
+        body: String? = nil
+    ) {
+        self.URI = URI
+        self.requestMethod = requestMethod
+        self.headers = headers
+        self.body = body
+    }
+
+    /// The URL of the request.
+    public let URI: URL
```

Comment:
Can we try to improve type-safety and update implementation to have something like that:

```swift
/// Returns the requested incoming header value.
func header(_ named: HTTPRequestHeader.Name) -> String?

(or at least to use `HTTPRequestHeader.Name` type as a subscript type?)

/// Add a header to the response.
/// No check for duplicate or repeated headers will be made.
func addHeader(_ named: HTTPRequestHeader.Name, value: String)

/// Set the indicated header value.
/// If the header already exists then the existing value will be replaced.
func setHeader(_ named: HTTPRequestHeader.Name, value: String)

/// Provide access to all current header values.
var headers: AnyIterator<(HTTPRequestHeader.Name, String)> { get }
```

?

Reply:
The same for body.

Reply:
We might want to use these method to avoid a huge initializers and make API clear.

WDYT?

Reply:
I've added method for headers. I suggest to resolve questions with `body` in the future if any.